hrms/payroll/doctype/additional_salary/additional_salary.py

```diff
@@ -117,16 +117,21 @@ def validate_duplicate_additional_salary(self):
 		if not self.overwrite_salary_structure_amount:
 			return
 
+		filters = {
+			"name": ["!=", self.name],
+			"salary_component": self.salary_component,
+			"payroll_date": self.payroll_date,
+			"overwrite_salary_structure_amount": 1,
+			"employee": self.employee,
+			"docstatus": 1,
+		}
+
+		if self.is_recurring:
+			filters["payroll_date"] = ["between", [self.from_date, self.to_date]]
```

Comment:
This should actually also check for the other case. If my current additional salary is not recurring but there is a recurring additional salary already existing.

Reply:
@ruchamahabal I believe it is already covered in the method before this call. Are you referring to a different case?

https://github.com/frappe/hrms/blob/341364cef473231f3acf0d3695d766510577182b/hrms/payroll/doctype/additional_salary/additional_salary.py#L26

Reply:
@ruchamahabal kind reminder